sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/LayoutTableWithSemanticMarkupCheck.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarHTML :: SonarQube Plugin
+ * Copyright (c) 2010-2019 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.sonar;
+
+import static java.lang.String.format;
+
+import java.util.Optional;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S5258")
+public class LayoutTableWithSemanticMarkupCheck extends AbstractPageCheck {
+
+  private boolean isWithinLayoutTable = false;
```

Comment:
I think you should use a `Deque<Boolean>` to keep whether last `table` element seen was a layout table in this case, otherwise we might have issue with nested tables.
Current implementation will have the following FP/FN:
```
<table role="presentation">
  <caption></caption> <!-- Noncompliant -->
  <tr>
    <td>first</td>
    <td>
      <table>
        <caption></caption> <!-- False positive, this caption is not related to a layout table -->
        <tr><td>nested table</td></tr>
        <tr><td>nested table</td></tr>
      </table>
    </td>
  </tr>
  <tr>
    <td></td>
    <td headers="h"></td> <!-- False negative -->
    <td scope="s"></td> <!-- False negative -->
  </tr>
</table>
```